rasa/nlu/training_data/formats/markdown.py

```diff
@@ -211,23 +211,31 @@ def dumps(self, training_data: "TrainingData") -> Text:
     def _generate_training_examples_md(self, training_data: "TrainingData") -> Text:
         """Generates markdown training examples."""
 
-        # Sort training data by intent
-        training_examples = sorted(
```

Comment:
I mean I see where this is coming from: the user doesn't sort the training data and when Rasa X dumps it it gets sorted. but I do not how the intent order is preserved in this case either, the new implementation
- relies on pythons internal dict ordering
- relies on the order of training examples, I am not sure that is necessarily preserved between reading and dumping through the database

Reply:
> relies on the order of training examples, I am not sure that is necessarily preserved between reading and dumping through the database

It is. They are inserted one after the other, and then sorted by id when retrieving them.

> relies on pythons internal dict ordering

The internal dict ordering is the insert order, what speaks against utilizing it?

Reply:
use orderrdict + write test for retrieval actions dumping